ribs/emitters/_gradient_emitter.py

```diff
@@ -0,0 +1,297 @@
+                size=(self.batch_size, self.solution_dim),
+            ).astype(self.archive.dtype)
+
+            sol = np.minimum(np.maximum(parents + noise, self.lower_bounds),
+                             self.upper_bounds)
+
+        self._parents = sol
+        return self._parents
+
+    def ask(self):
+        """Samples new solutions from a gradient arborescence parameterized by a
+        multivariate Gaussian distribution.
+
+        The multivariate Gaussian is parameterized by sigma_g.
+
+        This method returns ``batch_size`` solutions, even though one solution
+        is returned via ``ask_dqd``.
+        
+        Returns:
+            (:attr:`batch_size`, :attr:`solution_dim`) array -- a batch of new
+            solutions to evaluate.
+        Raises:
+            RuntimeError: This method was called without first passing gradients
+                with calls to ask_dqd() and tell_dqd().
+        """
+        if self.archive.empty and self._initial_solutions is not None:
+            return self._initial_solutions
+
+        if self._jacobian_batch is None:
+            raise RuntimeError("Please call ask_dqd() and tell_dqd() "
+                               "before calling ask().")
+
+        if self._measure_gradients:
+            noise = self._rng.normal(
+                loc=0.0,
+                scale=self._sigma_g,
+                size=self._jacobian_batch.shape[:2],
+            )
+            noise[:, 0] = np.abs(noise[:, 0])
+            noise = np.expand_dims(noise, axis=2)
+            offsets = np.sum(np.multiply(self._jacobian_batch, noise), axis=1)
+            sols = offsets + self._parents
+        else:
+            # Transform the Jacobian
+            if len(self._jacobian_batch.shape) == 3:
+                self._jacobian_batch = np.squeeze(self._jacobian_batch[:,
+                                                                       0:1, :],
+                                                  axis=1)
+            sols = self._parents + self._jacobian_batch * self._sigma_g
+
+        return sols
+
+    def tell_dqd(self,
+                 solution_batch,
+                 objective_batch,
+                 measures_batch,
+                 jacobian_batch,
+                 status_batch,
+                 value_batch,
+                 metadata_batch=None):
+        """Gives the emitter results from evaluating the solutions.
+
+        Args:
+            solution_batch (array-like): (batch_size, :attr:`solution_dim`)
+                array of solutions generated by this emitter's
+                :meth:`ask_dqd()` method.
+            objective_batch (array-like): 1d array containing the objective
+                function value of each solution.
+            measures_batch (array-like): (batch_size, measure space dimension)
+                array with the measure space coordinates of each solution.
+            jacobian_batch (array-like): (batch_size, 1 + measure_dim,
```

Comment:
I think we should allow Jacobian batch to be (batch_size, 1, solution_dim) if measure gradients are turned off.